aibtc-v1/chat.py

```diff
@@ -0,0 +1,171 @@
+import streamlit as st
+import logging
+from typing import Dict, Any
+from utils.session import init_session_state
+
+# Initialize session state
+init_session_state()
+
+# Set up Streamlit page
+st.set_page_config(page_title="CrewAI Chatbot", layout="centered")
+
+# Constants
+AVAILABLE_CREWS: Dict[str, Dict[str, Any]] = st.session_state.crew_mapping
+
+
+def load_custom_styles():
+    st.markdown(
+        """
+    <style>
+    /* Custom styles here */
+    </style>
+    """,
+        unsafe_allow_html=True,
+    )
+
+
+def generate_initial_welcome_message() -> str:
+    message = "Welcome to CrewAI Chatbot! I can help you run different CrewAIs. Here are the available options:\n\n"
+    message += "\n".join(
+        f"- **{name}**: {info['description']}" for name, info in AVAILABLE_CREWS.items()
+    )
+    message += "\n\nYou can ask me more about any of these or tell me what you want to do, and I'll recommend a CrewAI to run."
```

Comment:
```suggestion
    message += "\n\nYou can ask me more about any of these or tell me what you want to do, and I'll recommend a crew to run."
```

Also we can leverage a tool feature here if there's trouble in it decided to tell more about it or select one to run.

If we create a tool that is something like `provide_description(description_of_tool: str)` then we can set a feature to `return_direct` somewhere in the tool definition. That way if the LLM decides to describe something, it will reach for the tool, send the output, and that can be returned directly to the interface.

Otherwise we're just hoping it gets to final answer.